code/modules/mining/fulton.dm

```diff
@@ -0,0 +1,168 @@
+	for(var/B in total_extraction_beacons)
+		var/obj/machinery/extraction_point/EP = B
+		if(EP.beacon_network in beacon_networks)
+			possible_beacons += EP
+
+	if(!possible_beacons.len)
+		user << "There are no extraction beacons in existance!"
+		return
+
+	else
+		var/A
+
+		A = input("Select a beacon to connect to", "Balloon Extraction Pack", A) in possible_beacons
+
+		if(!A)
+			return
+		beacon = A
+
+/obj/item/weapon/extraction_pack/afterattack(atom/movable/A, mob/living/carbon/human/user, flag, params)
+	if(!beacon)
+		user << "[src] is not linked to a beacon, and cannot be used."
+		return
+	if(!can_use_indoors)
+		var/area/area = get_area(A)
+		if(!area.outdoors)
+			user << "[src] can only be used on things that are outdoors!"
+			return
+	if(!flag)
+		return
+	if(!istype(A))
+		return
+	else
+		if(!safe_for_living_creatures && check_for_living_mobs(A))
+			user << "[src] is not safe for use with living creatures, they wouldn't survive the trip back!"
+		if(A.loc == user || A == user) // no extracting stuff you're holding in your hands/yourself
+			return
+		if(A.anchored)
+			return
+		user << "<span class='notice'>You start attaching the pack to [A]...</span>"
+		if(do_after(user,50,target=A))
+			user << "<span class='notice'>You attach the pack to [A] and activate it.</span>"
+			uses_left--
+			if(uses_left <= 0)
+				user.drop_item(src)
+				loc = A
+			var/image/balloon
+			var/image/balloon2
+			var/image/balloon3
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				M.Weaken(16) // Keep them from moving during the duration of the extraction
+				M.buckled = 0 // Unbuckle them to prevent anchoring problems
+			else
+				A.anchored = 1
+				A.density = 0
+			var/obj/effect/extraction_holder/holder_obj = new(A.loc)
+			holder_obj.appearance = A.appearance
+			A.loc = holder_obj
+			balloon2 = image('icons/obj/fulton_balloon.dmi',"fulton_expand")
+			balloon2.pixel_y = 10
+			balloon2.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			holder_obj.overlays += balloon2
+			sleep(4)
+			balloon = image('icons/obj/fulton_balloon.dmi',"fulton_balloon")
+			balloon.pixel_y = 10
+			balloon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			holder_obj.overlays -= balloon2
+			holder_obj.overlays += balloon
+			playsound(holder_obj.loc, 'sound/items/fulext_deploy.wav', 50, 1, -3)
+			animate(holder_obj, pixel_z = 10, time = 20)
+			sleep(20)
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
+			playsound(holder_obj.loc, 'sound/items/fultext_launch.wav', 50, 1, -3)
+			animate(holder_obj, pixel_z = 1000, time = 30)
+			if(istype(A, /mob/living/carbon/human))
+				var/mob/living/carbon/human/L = A
+			//	L.forcesay(extraction_appends) // ";HELP BL-AAAAAAAAAAAAAAAAHHHHHHHHH"
```

Comment:
Gonna be honest, I've attempted to make code that does this gradual step by step stuff using sleeps() and all the maintainers commented that you have to check that the object/mob/whatever hasn't inexplicably died during these breaks. Every time.